### PR TITLE
Use 'tenant-id' instead of 'common' in ClientRegistration.

### PR DESCRIPTION
**Pull request by chenrujun**
Use 'tenant-id' instead of 'common' in ClientRegistration.

### CODE REVIEW

File: sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADOAuth2AutoConfiguration.java [hunk 1]
```diff
@@ -36,29 +42,69 @@
...
+    public ClientRegistrationRepository clientRegistrationRepository() {
+        return new InMemoryClientRegistrationRepository(azureClientRegistration());
+    }
+
+    private ClientRegistration azureClientRegistration() {
+        String tenantId = aadAuthenticationProperties.getTenantId().trim();
+        Assert.hasText(tenantId, "azure.activedirectory.tenant-id should have text.");
+        Assert.doesNotContain(tenantId, " ", "azure.activedirectory.tenant-id should not contain ' '.");
+        Assert.doesNotContain(tenantId, "/", "azure.activedirectory.tenant-id should not contain '/'.");
+        return ClientRegistration.withRegistrationId("azure")
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADOAuth2AutoConfiguration.java:74]: Will multi-tenant application still function with this change?

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADOAuth2AutoConfiguration.java (hunk 1 above)
Comment (chenrujun) [reply sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADOAuth2AutoConfiguration.java:74]: IMU, if we set `azure.activedirectory.tenant-id=common`, then multi-tenant application will function.

**Comment by chenrujun** [comment]
Hi, @saragluna , @yiliuTo .

Please review the PR when you have time. 🙏

File: sdk/spring/azure-spring-boot-starter-active-directory-b2c/pom.xml [hunk 2]
```diff
@@ -36,6 +36,11 @@
       <artifactId>azure-spring-boot</artifactId>
       <version>3.0.0-beta.1</version> <!-- {x-version-update;com.azure:azure-spring-boot;current} -->
     </dependency>
+    <dependency>
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-starter-active-directory-b2c/pom.xml:39]: The azure-identity is not used by the aad-b2c starter?

File: sdk/spring/azure-spring-boot-starter-active-directory/pom.xml [hunk 3]
```diff
@@ -35,6 +35,11 @@
       <artifactId>azure-spring-boot</artifactId>
       <version>3.0.0-beta.1</version> <!-- {x-version-update;com.azure:azure-spring-boot;current} -->
     </dependency>
+    <dependency>
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-starter-active-directory/pom.xml:38]: The aad starter does not use this azure-identity?

File: sdk/spring/azure-spring-boot-starter-servicebus-jms/pom.xml [hunk 4]
```diff
@@ -25,6 +25,11 @@
       <artifactId>azure-spring-boot</artifactId>
       <version>3.0.0-beta.1</version> <!-- {x-version-update;com.azure:azure-spring-boot;current} -->
     </dependency>
+    <dependency>
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-starter-servicebus-jms/pom.xml:28]: Service bus jms has no dependency on this.

**Review by saragluna — CHANGES_REQUESTED** [review]

**Review by saragluna — APPROVED** [review]
Let's merge this PR now and fix the useless dependencies in another PR.

**Pull request merged**